sympy/sets/sets.py

```diff
@@ -153,6 +153,181 @@ def isdisjoint(self, other):
         """
         return self.is_disjoint(other)
 
+    def _union_simplify(self, other):
```

Comment:
Since this method is not general ( it can simplify only specific types of union), I think all of this code should be moved to `ImageSet` class in sets/fancysets.py .